malariagen_data/ag3.py

```diff
@@ -6049,6 +6051,28 @@ def plot_snps(
 
         return fig
 
+    def aim_sites(self, aims):
+        """Open ancestry informative marker sites.
+
+        Parameters
+        ----------
+        aims : {'gamb_vs_colu', 'gambcolu_vs_arab'}
+
+        Returns
+        -------
+        ds : xarray.Dataset
+            A dataset containing aim positions and discriminating alleles.
+
+        """
+        try:
+            ds = self._cache_aims[aims]
+        except KeyError:
+            path = f"{self._base_path}/reference/aim_defs/{aims}.zarr"
+            store = init_zarr_store(fs=self._fs, path=path)
+            ds = xr.open_zarr(store, concat_characters=False)
+            self._cache_aims[aims] = ds
+        return ds
```

Comment:
There is a potential issue here with the caching of an xarray Dataset. If the user ever modified the dataset, e.g., by adding or removing variables, those changes would affect the cached dataset, and so a later call to `aim_sites()` would return the modified dataset. This is analogous to situations where a pandas DataFrame is cached (#184), where the resolution was to always return copies of any cached dataframes (#194).

 I would suggest doing something similar here. A nuance is I think we would get away with a shallow copy, because the actual variables are not writeable. I.e., replace:

```
return ds
```

...with:

```
return ds.copy(deep=False)
```